### PR TITLE
Look for clang++ on the OpenCL backend instead of HCC

### PR DESCRIPTION
**Pull request by daniellowell**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -122,15 +122,23 @@ add_definitions("-DHIP_COMPILER_FLAGS=${HIP_COMPILER_FLAGS}")
 if( MIOPEN_BACKEND STREQUAL "OpenCL")
     set(MIOPEN_BACKEND_OPENCL 1)
     find_package( OpenCL REQUIRED )
-    find_program(MIOPEN_HIP_COMPILER hcc
+    find_program(MIOPEN_HIP_COMPILER clang++
         PATH_SUFFIXES bin
-        PATHS /opt/rocm
+        PATHS /opt/rocm/llvm
```
Comment (pfultz2) [inline CMakeLists.txt:127]: This should probably be `HINTS` instead of `PATHS`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -99,7 +99,7 @@ else()
     set(WINE_CMD)
 endif()
 
-list(APPEND CMAKE_PREFIX_PATH /opt/rocm /opt/rocm/hcc /opt/rocm/hip)
+list(APPEND CMAKE_PREFIX_PATH /opt/rocm /opt/rocm/llvm /opt/rocm/hcc /opt/rocm/hip)
```
Comment (pfultz2) [inline CMakeLists.txt:102]: Is it necessary to add /opt/rocm/llvm to `CMAKE_PREFIX_PATH`?

File: src/hip/hip_build_utils.cpp [hunk 3]
```diff
@@ -96,7 +96,7 @@ boost::filesystem::path HipBuild(boost::optional<TmpDir>& tmp_dir,
     WriteFile(src, tmp_dir->path / filename);
     if(IsHccCompiler())
     {
-        params += " -amdgpu-target=" + dev_name;
+        params += " --std=c++11 -amdgpu-target=" + dev_name;
```
Comment (pfultz2) [inline src/hip/hip_build_utils.cpp:99]: Is this change really needed? This can prevent using newer c++'s. The kernel should pass the c++ version as part of the build flags.

File: hip-clang.docker [hunk 4]
```diff
@@ -46,7 +43,7 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-
 
 
 # Patch hip-clang
-ADD cmake/hip-config.cmake /opt/rocm/hip/lib/cmake/hip/hip-config.cmake
+#ADD cmake/hip-config.cmake /opt/rocm/hip/lib/cmake/hip/hip-config.cmake
```
Comment (pfultz2) [inline hip-clang.docker:46]: I think this can be completely deleted.

File: CMakeLists.txt (hunk 2 above)
Comment (daniellowell) [reply CMakeLists.txt:102]: It was not picking up clang++ over hcc even though I asked cmake to look for clang++ first. 
Though now it is making our HCC builds fail because it is picking up the system installed clang++.
http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/lookForClang/2/pipeline
I can remove it, but do you have suggestions on making sure the path gets picked up?

File: src/hip/hip_build_utils.cpp (hunk 3 above)
Comment (daniellowell) [reply src/hip/hip_build_utils.cpp]: Reverted.

File: hip-clang.docker (hunk 4 above)
Comment (daniellowell) [reply hip-clang.docker]: Deleted

File: CMakeLists.txt (hunk 1 above)
Comment (daniellowell) [reply CMakeLists.txt]: Why `HINTS`?

File: CMakeLists.txt [hunk 5]
```diff
@@ -108,35 +108,49 @@ set( MIOPEN_BACKEND ${MIOPEN_DEFAULT_BACKEND} CACHE STRING
...
+	        PATHS /opt/rocm/llvm
+        )
+        if(MIOPEN_HIP_COMPILER)
+            message("hip compiler: ${MIOPEN_HIP_COMPILER}")
+        else()
+            message(FATAL_ERROR "hip compiler not found")
+        endif()
     endif()
 
+    set (HIP_CXX_COMPILER ${MIOPEN_HIP_COMPILER})
```
Comment (atamazov) [inline CMakeLists.txt:135]: :anchor:
